src/utils/primary-action-handler.js

```diff
@@ -0,0 +1,132 @@
+export default class PrimaryActionHandler {
+  constructor(scene) {
+    this.scene = scene;
+    this.cursor = null;
+    this.isCursorInteracting = false;
+
+    this.registerCursor = this.registerCursor.bind(this);
+    this.isReady = this.isReady.bind(this);
+    this.addEventListeners = this.addEventListeners.bind(this);
+    this.onPrimaryDown = this.onPrimaryDown.bind(this);
+    this.onPrimaryUp = this.onPrimaryUp.bind(this);
+    this.onGrab = this.onGrab.bind(this);
+    this.onRelease = this.onRelease.bind(this);
+    this.onCardboardButtonDown = this.onCardboardButtonDown.bind(this);
+    this.onCardboardButtonUp = this.onCardboardButtonUp.bind(this);
+    this.onMoveDuck = this.onMoveDuck.bind(this);
+  }
+
+  registerCursor(cursor) {
+    this.cursor = cursor;
+    if (this.isReady()) {
+      this.addEventListeners();
+    }
+  }
+
+  isReady() {
+    return this.cursor;
+  }
+
+  addEventListeners() {
+    this.scene.addEventListener("action_primary_down", this.onPrimaryDown);
+    this.scene.addEventListener("action_primary_up", this.onPrimaryUp);
+    this.scene.addEventListener("action_grab", this.onGrab);
+    this.scene.addEventListener("action_release", this.onRelease);
+    this.scene.addEventListener("cardboardbuttondown", this.onCardboardButtonDown);
+    this.scene.addEventListener("cardboardbuttonup", this.onCardboardButtonUp);
+    this.scene.addEventListener("move_duck", this.onMoveDuck);
+  }
+
+  onMoveDuck(e) {
+    if (this.isCursorInteracting) {
+      this.cursor.changeDistanceMod(-e.detail.axis[1] / 8);
+    }
+  }
+
+  onGrab(e) {
+    if (e.target.id.match(this.cursor.data.handedness)) {
```

Comment:
This should just check if `e.target === this.cursor.controller`

Reply:
👍